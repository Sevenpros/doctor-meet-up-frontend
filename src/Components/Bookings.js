import React from 'react';
import { useSelector } from 'react-redux';
import Booking from './Booking';

const Bookings = () => {
  const bookings = useSelector((state) => state.bookingReducer);
  return (
    <div>
      {bookings.map((booking) => (<Booking key={booking.id} booking={booking} />))}
    </div>
  );
};
export default Bookings;
