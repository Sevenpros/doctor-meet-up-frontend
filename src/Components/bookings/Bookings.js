/* eslint-disable */

import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchBookings } from '../../Redux/bookings/bookings';
import SideNav from '../sideNav';
const Bookings = () => {
  const dispatch = useDispatch();
  const bookings = useSelector((state) => state.bookings);
  console.log(bookings);
  // const doctors = useSelector((state) => state.doctors.doctors);
  // const status = useSelector((state) => state.bookings.status);

  useEffect(() => {
    dispatch(fetchBookings());
    // dispatch(fetchDoctors());
  }, [dispatch]);
  const displayBookings = bookings.length !== 0 ? (
    bookings.map((booking) => (
      <div className="reserve-dev">
        <div key={booking.id}>
          {/**
            {doctors.map(
              (doctor) =>
                doctor.id === booking.doctor_id && (
                   )
            )}
            */}
            <strong>Doctor Name</strong>
            <h3>first_name : {booking.doctor.first_name}</h3>
            <h3>last_name: {booking.doctor.last_name}</h3>
               
          <h4>City: {booking.city}</h4>
          <p>Date: {booking.date}</p>
          <p>Time: {booking.time}</p>
          <p>Disease description: {booking.disease_desc}</p>
          <button
            className="btn btn-danger"
            onClick={(e) => {
              e.preventDefault();
              fetch(`http://localhost:3000/bookings/${booking.id}`, {
                method: 'DELETE',
                headers: {
                  'Content-Type': 'application/json',
                },
              });
              dispatch(fetchBookings());
            }}
          >
            Cancel Booking
          </button>
        </div>
      </div>
    ))
  ) : (
    <div>
      <h4 className="reservations">No bookings.</h4>
    </div>
  );

  return (
    <div>
      <SideNav />
      <h1>BOOKINGS LIST</h1>
      {displayBookings }
      
    </div>
  );
};

export default Bookings;
