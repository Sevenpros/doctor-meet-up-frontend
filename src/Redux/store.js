import { configureStore } from '@reduxjs/toolkit';
// import logger from 'redux-logger';
import doctorsReducer from './doctors/doctors';
import doctorReducer from './doctors/doctor';

const store = configureStore({
  reducer: {
    doctors: doctorsReducer,
    doctor: doctorReducer,
  },
  // middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});
export default store;
