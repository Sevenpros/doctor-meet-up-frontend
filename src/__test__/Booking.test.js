import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';

import Bookings from '../Components/bookings/Bookings';  // eslint-disable-line

describe('Bookings tests', () => {
  it('render Bookings test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Bookings />
        </Router>
      </Provider>,

    );
    expect(tree).toMatchSnapshot();
  });
});
