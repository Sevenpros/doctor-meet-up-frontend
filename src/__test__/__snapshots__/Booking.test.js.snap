// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookings tests render Bookings test 1`] = `null`;
